OpenProblemLibrary/Hope/Multi1/03-05-Basis-subspace/Basis_30.pg

```diff
@@ -47,6 +40,47 @@ $rrefA = Matrix([
 [0,0,0,0]
 ]);
 
+
+$P = Matrix([[1,0,0], [0,1,0], [0, 0, 1]]);
+
+$paddedA = $P*$rrefA;
+
+Context()->variables->are(
+'a1'=>['Vector3D','TeX'=>'\mathbf{a_1}'],
+'a2'=>['Vector3D','TeX'=>'\mathbf{a_2}'],
+'a3'=>['Vector3D','TeX'=>'\mathbf{a_3}'],
+'a4'=>['Vector3D','TeX'=>'\mathbf{a_4}'],
+);
+
+
+$list_checker = sub {
+  my ($correct, $student, $ansHash, $value) = @_;
+
+  # pull the column vectors into variables
+  my ($a1,$a2,$a3,$a4) = (
+    Vector($rrefA->column(1)),
+    Vector($rrefA->column(2)),
+    Vector($rrefA->column(3)),
+    Vector($rrefA->column(4))
+  );
+  
+  # loop through the student list and parse any formulas in the student response
+  my $parsed_student = [];
+  foreach my $el (@$student) {
+    my $parsed_el = $el;
+    if ($el->class eq 'Formula') {
+      $parsed_el = $el->eval(a1=>$a1,a2=>$a2,a3=>$a3,a4=>$a4);
+    } else {
+        return 0;
+    }
```

Comment:
So students are _required_ to list formulas then? The instructions still say, "Each vector should have the form [| <a,b,c> |] or [| <a,b,c,d> |] where a,b,... are numbers" but this is no longer true.

Reply:
Indeed, the requirement that explicit vectors be entered should be removed. My bad.

The answer checker will still accept explicit vectors. It's just that they'd be marked incorrect.